dom/events/Event-dispatch-detached-input-and-change.html

```diff
@@ -0,0 +1,226 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<link rel="author" title="Joey Arhar" href="mailto:jarhar@chromium.org">
+<title>input and change events for detached checkbox and radio elements</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<body>
+<script>
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.click();
+  assert_false(inputEventFired);
+  assert_false(changeEventFired);
+
+  t.done();
+}, 'detached checkbox should not emit input or change events on click().');
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'radio';
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.click();
+  assert_false(inputEventFired);
+  assert_false(changeEventFired);
+
+  t.done();
+}, 'detached radio should not emit input or change events on click().');
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.dispatchEvent(new MouseEvent('click'));
+  assert_false(inputEventFired);
+  assert_false(changeEventFired);
+
+  t.done();
+}, `detached checkbox should not emit input or change events on dispatchEvent(new MouseEvent('click')).`);
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'radio';
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.dispatchEvent(new MouseEvent('click'));
+  assert_false(inputEventFired);
+  assert_false(changeEventFired);
+
+  t.done();
+}, `detached radio should not emit input or change events on dispatchEvent(new MouseEvent('click')).`);
+
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+  document.body.appendChild(input);
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.click();
+  assert_true(inputEventFired);
+  assert_true(changeEventFired);
+
+  input.remove();
```

Comment:
To make cleanup happen even if the test fails, you should move this up to line 76 and wrap it in `t.add_cleanup(() => input.remove())`.

However, it's also fairly unnecessary. A few extra elements in the DOM at the end of the test doesn't hurt anyone. So you could also just remove these.

Reply:
I removed the cleanup